src/transports/ssh.c

```diff
@@ -580,6 +580,24 @@ static int _git_ssh_setup_conn(
 				case LIBSSH2_HOSTKEY_TYPE_DSS:
 					cert.raw_type = GIT_CERT_SSH_RAW_TYPE_DSS;
 					break;
+					
+#ifdef LIBSSH2_HOSTKEY_TYPE_ECDSA_256
+				case LIBSSH2_HOSTKEY_TYPE_ECDSA_256:
+					cert.raw_type = GIT_CERT_SSH_RAW_TYPE_KEY_ECDSA_256;
+					break;
+				case LIBSSH2_HOSTKEY_TYPE_ECDSA_384:
+					cert.raw_type = GIT_CERT_SSH_RAW_TYPE_KEY_ECDSA_384;
+					break;
+				case LIBSSH2_KNOWNHOST_KEY_ECDSA_521:
```

Comment:
is it possible for only a subset of these to be defined? (say, only 256 but not 384 or 521). or were all of these constants introduced in the same commit?

Reply:
Yeah, 256, 384 and 521 were all introduced at the same time in 62b825c8.
ED25519 was added a bit later with 030922925.